Backend/app/rider-platform/rider-app/Main/src/Beckn/OnDemand/Utils/Init.hs

```diff
@@ -90,8 +90,15 @@ mkStops origin mDestination mStartOtp intermediateStops =
           <> (map (\(location, order) -> UCommon.mkIntermediateStop location order (order - 1)) $ zip intermediateStops [1 ..])
 
 mkPayment :: Maybe DMPM.PaymentMethodInfo -> DBC.BecknConfig -> Context.City -> [Spec.Payment]
-mkPayment _ bapConfig city = do
-  let mkParams :: (Maybe BknPaymentParams) = decodeFromText =<< bapConfig.paymentParamsJson
+mkPayment mbPaymentMethodInfo bapConfig city = do
+  let mkParams :: (Maybe BknPaymentParams) =
+        case mbPaymentMethodInfo of
+          Just paymentMethodInfo ->
+            -- add this part only for testing purpose
+            if paymentMethodInfo.paymentInstrument == DMPM.Cash
+              then Nothing
+              else Just $ BknPaymentParams Nothing Nothing (Just $ show paymentMethodInfo.paymentInstrument) -- Bpp side we consider online when vpa is present (Ideally we should put upi_id here)
```

Comment:
On bpp side you used payment in `confirm` api, but did not added on rider app
Also you can reuse duplicated part